agent/exploits/cve_2024_34470.py

```diff
@@ -0,0 +1,32 @@
+"""Agent Asteroid implementation for CVE-2024-34470"""
+
+import re
+from packaging import version
+from agent.exploits import webexploit
+from agent import exploits_registry
+from agent import definitions
+
+VULNERABILITY_TITLE = "HSC MAILINSPECTOR PATH TRAVERSAL"
+VULNERABILITY_REFERENCE = "CVE-2024-34470"
+VULNERABILITY_DESCRIPTION = """An issue was discovered in HSC Mailinspector 5.2.17-3 through v.5.2.18.
+An Unauthenticated Path Traversal vulnerability exists in the /public/loader.php file. The path parameter does not properly filter whether the file and directory passed are part of the webroot, allowing an attacker to read arbitrary files on the server.
+"""
+RISK_RATING = "CRITICAL"
+MIN_VULNERABLE_VERSION = version.parse("5.2.17-3")
+MAX_VULNERABLE_VERSION = version.parse("5.2.18")
+
+
+@exploits_registry.register
+class CVE202434470Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(method="GET", path="/mailinspector/login.php")
+    check_request = definitions.Request(method="GET", path="/mailinspector/login.php")
+    accept_pattern = [re.compile("HSC MailInspector")]
+    vuln_ranges = [
+        definitions.VulnRange(MIN_VULNERABLE_VERSION, MAX_VULNERABLE_VERSION)
+    ]
```

Comment:
I think the rule should define the version pattern to tell where to extract it. There are many cases where many versions will be in the response, like jquery for instance, and that can be false positive prone.

Reply:
fixed